test/e2e/serial/tests/workload_placement.go

```diff
@@ -276,10 +280,21 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload placeme
 				return fxt.Client.Update(context.TODO(), updatedDp)
 			}).WithTimeout(2 * time.Minute).WithPolling(10 * time.Second).ShouldNot(HaveOccurred())
 
+			// wait for the replicasets under the deployment are increased by 1
+			namespacedDpName := fmt.Sprintf("%s/%s", updatedDp.Namespace, updatedDp.Name)
+			Eventually(func() int {
+				r, err := replicasets.GetReplicasetByDeployment(fxt.Client, context.TODO(), *updatedDp)
+				Expect(err).ToNot(HaveOccurred())
+				klog.Infof("number of replicasets under deployment %q is %d", namespacedDpName, len(r))
+				return len(r)
+			}).WithTimeout(10*time.Second).WithPolling(1*time.Second).Should(Equal(len(dpReplicas)+1), "deployment %q replicasets were not increased by 1", namespacedDpName)
+
+			dpReplicas, err = replicasets.GetReplicasetByDeployment(fxt.Client, context.TODO(), *updatedDp)
```

Comment:
Can we simplify this check just checking the related Deployment is ready? If not fine, but please document why.

Reply:
I don't mind adding extra utility functions (if their use is warranted), but in the test code I'd rather spend more time understanding what's going on. Either we are checking the wrong condition, or there's more going on and we're are kinda sidestepping the root cause.